src/core/views.py

```diff
@@ -1785,10 +1801,26 @@ def serve_file_one_click(
 
     try:
         fsock = open(file_path, 'r')
-        mimetype = mimetypes.guess_type(file_path)
+
+        mimetype_guess = mimetypes.guess_type(file_path)
+        try:
+            mimetype = mimetype_guess[0]
+            if not mimetype:
+                mimetype = 'unknown'
+        except IndexError:
+            mimetype = 'unknowns'
+
         response = StreamingHttpResponse(fsock, content_type=mimetype)
+
+        # Strip non-ascii characters from filename
+        # to avoid a bug where an unusable ZIP file is served
         response['Content-Disposition'] = (
-                "attachment; filename=%s" % _file.original_filename
+            "attachment; filename={file_name}".format(
+                file_name=_file.original_filename.encode(
+                    'ascii',
+                    errors='ignore'
+                )
+            )
```

Comment:
If you have any feeling that we will have to guess the mimetype again and create a `Content-Disposition`  again, we might need to move both to a dedicated function we can reuse. If that is not the case, we can leave as it is.